save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/TopBar.kt

```diff
@@ -347,5 +279,81 @@ fun topBar() = FC<TopBarProps> { props ->
     }
 }
 
-private fun textColor(hrefAnchor: String, location: history.Location) =
+/**
+ * [Props] of the top bor component
+ */
+external interface TopBarProps : PropsWithChildren {
+    /**
+     * Currently logged in user or null
+     */
+    var userInfo: UserInfo?
+
+    /**
+     * true if the device is mobile (screen is less 1000px)
+     */
+    var isMobileScreen: Boolean?
+}
+
+/**
+ * [Props] of the top bor component with [location]
+ */
+external interface TopBarPropsWithLocation : TopBarProps {
+    /**
+     * Is location
+     */
+    var location: Location
+}
+
+/**
+ * @property hrefAnchor is link
+ * @property width is width of the link text
+ * @property text is link text
+ */
+data class TopBarLink(
+    val hrefAnchor: String,
+    val width: Width,
+    val text: String,
+)
+
+private fun ChildrenBuilder.dropdownEntry(
+    faIcon: FontAwesomeIconModule?,
+    text: String,
+    isSelected: Boolean = false,
+    handler: ChildrenBuilder.(ButtonHTMLAttributes<HTMLButtonElement>) -> Unit = { },
+) = button {
+    type = ButtonType.button
+    val active = if (isSelected) "active" else ""
+    className = ClassName("btn btn-no-outline dropdown-item rounded-0 shadow-none $active")
+    faIcon?.let {
+        fontAwesomeIcon(icon = faIcon) {
+            it.className = "fas fa-sm fa-fw mr-2 text-gray-400"
+        }
+    }
+    +text
+    handler(this)
+}
+
+/**
+ * A component for web page top bar
+ *
+ * @return a function component
+ */
+fun topBar() = FC<TopBarProps> { props ->
+    val location = useLocation()
+    nav {
+        className = ClassName("navbar navbar-expand navbar-dark bg-dark topbar mb-3 static-top shadow mr-1 ml-1 rounded")
+        id = "navigation-top-bar"
+        topBarUrlSplits {
+            this.location = location
+        }
+        topBarLinks {
+            this.location = location
+        }
+        topBarUserField {
+            userInfo = props.userInfo
+        }
```

Comment:
Move all those FCs to external files and you will get an approve

Reply:
Done

Reply:
Done but in a bad way, needs to be fixed